ISPN-15069 CVE-2023-4586 Hot Rod TLS Hostname validation (#11148)
infinispan/infinispan: documentation/src/main/asciidoc/topics/proc_configuring_server_keystores.adoc

{brandname} Server supports the following keystore formats: JKS, JCEKS, PKCS12/PFX and PEM.
BKS, BCFKS, and UBER are also supported if the https://www.bouncycastle.org/[Bouncy Castle] library is present.

> Certificates should include the `subjectAltName` extension of type `dNSName` and/or `iPAddress` so that clients can correctly perform
> hostname validation, according to the rules defined by the https://datatracker.ietf.org/doc/html/rfc2818[RFC 2818] specification.
> The server will issue a warning if it is started with a certificate which does not include such an extension.

Reviewer: ```suggestion
Certificates must include the `subjectAltName` extension of the `dNSName` or `iPAddress` type, or both, to enable clients to perform hostname validation, following the rules defined by the https://datatracker.ietf.org/doc/html/rfc2818[RFC 2818] specification.
If the certificate does not include this extension, the server issues a warning upon startup.
```
